src/test/e2e/specs/mocks/server.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import * as vscode from 'vscode';
+import { ZodiosPathsByMethod, ZodiosResponseByPath } from '@zodios/core/lib/zodios.types';
+import { ResponseResolver, http } from 'msw';
+import { setupServer, SetupServerApi } from 'msw/node';
+import { apiClient, TerraformCloudAPIUrl } from '../../../../api/terraformCloud';
+import { handlers } from './handlers';
+
+type Api = typeof apiClient.api;
+
+export function mockGetApiClient<Path extends ZodiosPathsByMethod<Api, 'get'>>(
+  path: Path,
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  resolver: ResponseResolver<{}, never, Awaited<ZodiosResponseByPath<Api, 'get', Path>>>,
+) {
+  return http.get(`${TerraformCloudAPIUrl}${path}`, resolver);
+}
+
+let server: SetupServerApi;
+export let debugChannel: vscode.OutputChannel;
+
+export function setupMockServer() {
+  debugChannel = vscode.window.createOutputChannel('MSW Debug Channel');
+
+  server = setupServer(...handlers);
+
+  server.events.on('request:unhandled', ({ request }) => {
+    debugChannel.appendLine(
+      `Intercepted a request without a matching request handler: ${request.method} ${request.url}`,
+    );
+  });
+
+  server.events.on('response:mocked', ({ request, response }) => {
+    debugChannel.appendLine(
+      `Outgoing request "${request.method} ${request.url}" received mock response: ${response.status} ${response.statusText}`,
+    );
+  });
+  server.listen();
+}
+
+export function stopMockServer() {
+  server.close();
+}
+
+export function appendLine(line: string) {
+  debugChannel?.appendLine(line);
+}
```

Comment:
I used it heavily getting this to work to log and figure out why OrganizationPicker was popping twice, but took them out when done because I didn't want to deal with possible undefined.

Realistically we need these lines to stay, but did not want to spend the time to create a more full featured usage at this time.